### PR TITLE
Community/DAO fraud detection

### PR DESCRIPTION
**Pull request by CAPtheorem**
1/ Simple docker-compose to run local DTL and Verifier
2/ TS Fraud-detector service for local detection of operator fraud

### CODE REVIEW

File: boba_community/fraud-detector/docker-detect.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+x-var: &DATA_TRANSPORT_LAYER__ADDRESS_MANAGER
+  DATA_TRANSPORT_LAYER__ADDRESS_MANAGER=0x8376ac6C3f73a25Dd994E0b0669ca7ee0C02F089
+
+x-var: &ETH1_CTC_DEPLOYMENT_HEIGHT
+  ETH1_CTC_DEPLOYMENT_HEIGHT=13011896
+
+services:
+
+  deployer:
```
Comment (boyuan-chen) [inline boba_community/fraud-detector/docker-detect.yml]: Please remove the deployer. I have opened the port of the replica. The address of `addresses.json` is `https://replica.boba.network:8081/addresses.json` and the address of the `state-dump.latest.json` is `https://replica.boba.network:8081/state-dump.latest.json`. Others look good!

**Review by boyuan-chen — APPROVED** [review]

File: boba_community/fraud-detector/src/exec/run.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    throw new Error('Must pass ADDRESS_MANAGER_ADDRESS')
+  }
+  if (!L1_NODE_WEB3_URL) {
+    throw new Error('Must pass L1_NODE_WEB3_URL')
+  }
+  if (!VERIFIER_WEB3_URL) {
+    throw new Error('Must pass VERIFIER_WEB3_URL')
+  }
+
+  const l2Provider = new providers.JsonRpcProvider(VERIFIER_WEB3_URL)
```
Comment (InoMurko) [inline boba_community/fraud-detector/src/exec/run.ts]: maybe use ethers.StaticJsonRpcProvider?

File: boba_community/fraud-detector/src/exec/run.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  }
+  if (!L1_NODE_WEB3_URL) {
+    throw new Error('Must pass L1_NODE_WEB3_URL')
+  }
+  if (!VERIFIER_WEB3_URL) {
+    throw new Error('Must pass VERIFIER_WEB3_URL')
+  }
+
+  const l2Provider = new providers.JsonRpcProvider(VERIFIER_WEB3_URL)  
+  const l1Provider  = new providers.JsonRpcProvider(L1_NODE_WEB3_URL)
```
Comment (InoMurko) [inline boba_community/fraud-detector/src/exec/run.ts]: same as above

File: boba_community/fraud-detector/src/exec/run.ts (hunk 2 above)
Comment (CAPtheorem) [reply boba_community/fraud-detector/src/exec/run.ts]: Yup - good catch

File: boba_community/fraud-detector/src/exec/run.ts (hunk 3 above)
Comment (CAPtheorem) [reply boba_community/fraud-detector/src/exec/run.ts]: ditto

File: boba_community/fraud-detector/docker-detect.yml (hunk 1 above)
Comment (CAPtheorem) [reply boba_community/fraud-detector/docker-detect.yml]: Done

**Review by InoMurko — APPROVED** [review]

**Pull request merged**